src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,195 @@
+    // .setAudience(Collections.singletonList(CLIENT_ID)).build();
+
+    // GoogleIdToken idToken = verifier.verify(idTokenString);
+    // if (idToken == null) {
+    //   throw new Exception("ID token verification failed");
+    // }
+
+    // payload = idToken.getPayload();
+  }
+
+  /**
+   * This creates a standard universal user instance regardless of login status.
+   * This test method will not be removed from the webapp prior to deployment.
+   */
+  public static User getTestUser() {
+    User testUser = new User();
+    String testID = "000000000000000000000";
+    Key userKey = KeyFactory.createKey("User", testID);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    try {
+        testUser.entity = datastore.get(userKey);
+    } catch(Exception e) {
+        // On EntityNotFoundException, create initial instance.
+        testUser.entity = new Entity(userKey);
+        String emptyJsonArray = "[ ]";
+        testUser.entity.setProperty("displayName", "Test Person");
+        testUser.entity.setProperty("cookbook", emptyJsonArray);
+        testUser.entity.setProperty("userRecipes", emptyJsonArray);
+        testUser.entity.setProperty("userRecipes", emptyJsonArray);
+        testUser.entity.setProperty("planner", emptyJsonArray);
+    }
+    return testUser;
+  }
+
+  public long getDisplayName() {
+    return (long)entity.getProperty("displayName");
+  }
+
+  public void setDisplayName(String name) {
+    entity.setProperty("displayName", name);
+    put();
+  }
+
+// Cookbook and Planner currently support only one type of Recipe entity (namely public).
+// Likewise, userRecipes should only include private recipes.
+
+  public ArrayList<Long> getCookbook() {
+    return getPropertyArrayList("cookbook");
   }
 
-  public void setId(String id) {
-    this.id = id;
+  public ArrayList<Long> getUserRecipes() {
+    return getPropertyArrayList("userRecipes");
   }
 
-  public boolean equalIds(String id) {
-    return this.id.equals(id);
+  public ArrayList<Long> getPlanner() {
+    return getPropertyArrayList("planner");
   }
 
   public void addRecipeToPlanner(long id) {
-    planner.add(id);
+    addIDToRecipeList(id, "planner");
   }
 
   public void addRecipeToCookbook(long id) {
-    cookbook.add(id);
+    addIDToRecipeList(id, "cookbook");
   }
 
   public void addRecipeToUserRecipes(long id) {
-    userRecipes.add(id);
+    addIDToRecipeList(id, "userRecipes");
   }
 
   public void removeRecipeFromPlanner(long id) {
-    planner.remove(new Long(id));
+    removeIDFromRecipeList(id, "planner");
   }
 
   public void removeRecipeFromCookbook(long id) {
-    cookbook.remove(new Long(id));
+    removeIDFromRecipeList(id, "cookbook");
   }
 
   public void removeRecipeFromUserRecipes(long id) {
-    userRecipes.remove(new Long(id));
+    removeIDFromRecipeList(id, "userRecipes");
+  }
+
+  /**
+   * @Param id recipe ID to be removed from current User entity.
```

Comment:
I think we're supposed to still include a brief description of the methods even if the name is pretty obvious

Reply:
https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory
So not generally the case, but I thought these two could use a tidbit.

Reply:
Sounds Good